pkg/api/validation/validation.go

```diff
@@ -372,6 +372,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceVersion"), newMeta.ResourceVersion, "must be specified for an update"))
 	}
 
+	// Generation should be the same or at most incremented by one.
+	if newMeta.Generation != oldMeta.Generation && newMeta.Generation != oldMeta.Generation+1 {
```

Comment:
We want to start using it in the client for Deployments as a minimal trigger of redeploying the same thing cc: @smarterclayton